drivers/sensor/shell_battery.c

```diff
@@ -54,13 +54,12 @@ static int cmd_battery(const struct shell *shell, size_t argc, char **argv)
 	struct sensor_value temp, volt, current, i_desired, charge_remain;
 	struct sensor_value charge, v_desired, v_design, cap, nom_cap;
 	struct sensor_value full, empty;
-	const struct device *dev;
+	const struct device *dev = DEVICE_DT_GET(DT_ALIAS(battery));
```

Comment:
should this be a chosen node instead of an alias?

Reply:
This looks like the classic "we have no way to iterate over all the devices which implement API `foo`", right?

Reply:
> This looks like the classic "we have no way to iterate over all the devices which implement API `foo`", right?

we need device classes :-)